### PR TITLE
Moving clientauth package to be under client

### PR DESCRIPTION
**Pull request by krousey**
Doing thing in preparation to version clients. This is where it belongs anyway.

@caesarxuchao @bgrant0607

### CODE REVIEW

**Comment by krousey** [comment]
@k8s-bot test this

**Comment by krousey** [comment]
Huh. I don't know how this could cause e2e test failures. It should only cause build failures if anything.

@k8s-bot test this

File: pkg/client/auth/clientauth.go [hunk 1]
```diff
@@ -60,7 +60,7 @@ Example:
     client := client.New(clientConfig)
     client.Pods(ns).List()
 */
-package clientauth
```
Comment (nikhiljindal) [inline pkg/client/auth/clientauth.go]: Line 50 and 53 should be changed to client.auth as well?

Comment (nikhiljindal) [inline pkg/client/auth/clientauth.go]: Also update the package name in line 18

**Comment by nikhiljindal** [comment]
Mostly LGTM, just 2 comments.

File: pkg/client/auth/clientauth.go (hunk 1 above)
Comment (krousey) [inline pkg/client/auth/clientauth.go]: Done.

**Comment by krousey** [comment]
Fixed. Thanks.

**Comment by nikhiljindal** [comment]
LGTM, thanks for the fixes!

**Comment by krousey** [comment]
@k8s-bot test this and please don't flake this time

**Pull request merged**